kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/client/DefaultFileSystemClient.java

```diff
@@ -87,8 +87,7 @@ public CloseableIterator<FileStatus> listFrom(String filePath) throws IOExceptio
     public String resolvePath(String path) throws IOException {
         Path pathObject = new Path(path);
         FileSystem fs = pathObject.getFileSystem(hadoopConf);
-        Path resolvedPath = fs.resolvePath(pathObject);
```

Comment:
This seems to be the problem earlier. We should call `resolvePath` that checks for path existances. Instead the goal of this API is to fully qualify the path which `fs.makeQualified` does.